ksqldb-functional-tests/src/test/resources/query-validation-tests/row-offset-and-partition.json

```diff
@@ -134,12 +134,142 @@
         "ksql.rowpartition.rowoffset.enabled": true
       },
       "statements": [
-        "CREATE STREAM INPUT (id int KEY, ROWPARTITION INTEGER, ROWOFFSET BIGINT) WITH (kafka_topic='test_topic', value_format='JSON');"
+        "CREATE STREAM input (id INT KEY, ROWPARTITION INTEGER, ROWOFFSET BIGINT) WITH (kafka_topic='test_topic', value_format='JSON');"
       ],
       "expectedException": {
         "type": "io.confluent.ksql.util.KsqlStatementException",
         "message": "'ROWPARTITION' is a reserved column name. You cannot use it as a name for a column."
       }
+    },
+    {
+      "name": "verify ROWPARTITION is not queryable with feature disabled",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": false
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, val INTEGER) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM output AS SELECT ROWPARTITION FROM input EMIT CHANGES;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "SELECT column 'ROWPARTITION' cannot be resolved."
+      }
+    },
+    {
+      "name": "verify ROWOFFSET is not queryable with feature disabled",
+      "properties": {
+        "ksql.rowpartition.rowoffset.enabled": false
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, val INTEGER) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM output AS SELECT ROWOFFSET FROM input EMIT CHANGES;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "SELECT column 'ROWOFFSET' cannot be resolved."
+      }
+    },
+    {
+      "name": "verify INSERT VALUES works for ROWPARTITION and ROWOFFSET with feature disabled (matching types)",
```

Comment:
This is not the right place to be testing INSERT VALUES: QTT does not support INSERT VALUES. Rather, INSERT VALUES is tested in RQTT instead (see the file named `insert-values.json`).

Also, I don't think we need tests for verifying that inserts into ROWPARTITION AND ROWOFFSET are not allowed if the feature flag is not enabled. If there were a bug here, it'd be because the feature flag is not set up properly. This is a good thing to test while introducing the feature flag, but it's not necessary to check in since we'd just remove it once we ungate the feature flag anyway.

Reply:
Incidentally, QTT _is_ the correct place to check that inserting into ROWPARTITION and ROWOFFSET are disabled for INSERT INTO statements (in contrast to INSERT VALUES statements). It doesn't have to be in this PR, but we should make sure that ROWPARTITION and ROWOFFSET are properly disabled for INSERT INTO, and add the relevant test coverage. (Can you check that we have a ticket capturing this, and add one if not?)